Source/JavaScriptCore/wasm/WasmTypeDefinition.cpp

```diff
@@ -260,6 +303,39 @@ struct StructParameterTypes {
     }
 };
 
+struct ArrayParameterTypes {
+    FieldType elementType;
+
+    static unsigned hash(const ArrayParameterTypes& params)
+    {
+        return computeArrayTypeHash(params.elementType);
+    }
+
+    static bool equal(const TypeHash& sig, const ArrayParameterTypes& params)
+    {
+        if (!sig.key->is<ArrayType>())
+            return false;
+
+        const ArrayType* arrayType = sig.key->as<ArrayType>();
+
+        if (arrayType->elementType() != params.elementType)
+            return false;
+
+        return true;
+    }
+
+    static void translate(TypeHash& entry, const ArrayParameterTypes& params, unsigned)
```

Comment:
What does this function do?

Reply:
It's used to actually construct the hash set entry when it's needed, i.e., if the hash set has determined that the entry is not already in the set. If it's already in the set, the user of the set can avoid constructing the entry.

(relevant source code comment https://searchfox.org/wubkat/source/Source/WTF/wtf/HashSet.h#105)